Frameworks/UIKit/StarboardXaml/StarboardXaml.cpp

```diff
@@ -144,23 +145,37 @@ void UIApplicationActivated(IActivatedEventArgs^ args) {
     }
 
     if (args->Kind == ActivationKind::ToastNotification) {
-        Platform::String^ argsString = safe_cast<ToastNotificationActivatedEventArgs^>(args)->Argument;
-        TraceVerbose(TAG, L"Received toast notification with argument - %s", argsString->Data());
+        ToastNotificationActivatedEventArgs^ toastArgs = safe_cast<ToastNotificationActivatedEventArgs^>(args);
+        TraceVerbose(TAG, L"Received toast notification with argument - %ls", toastArgs->Argument->Data());
+
+        // Create list of keys so we can modify the valueset
+        std::vector<Platform::String^> keys;
+        for (auto i = toastArgs->UserInput->First(); i->HasCurrent; i->MoveNext()) {
+            keys.push_back(i->Current->Key);
+        }
+        
+        // Convert String^ to IPropertyValue so they can be converted to NSString*
+        for (auto i : keys) {
+            Platform::Object^ value = toastArgs->UserInput->Lookup(i);
+            toastArgs->UserInput->Insert(i, PropertyValue::CreateString(static_cast<Platform::String^>(value)));
```

Comment:
The reference is read only, but we can still replace the values like this.  It's not good, but it's the only way to change the String^ to IPropertyValue^

---
In reply to: [73012825](https://github.com/Microsoft/WinObjC/pull/637#discussion_r73012825) [](ancestors = 73012825,72850903,72745061)